enso/experiment/__init__.py

```diff
@@ -112,15 +116,16 @@ def experiment_has_been_run(self, current_settings):
         return True
 
     def _run_sub_experiment(
-        self, experiment_cls, dataset, train, test, target, current_setting
+        self, experiment_cls, dataset, train, test, target, current_setting, experiment_hparams={}
```

Comment:
I would do 

```...experiment_hparams=None):

experiment_hparams = experiment_hparams or None
```

to avoid accidentally mutating the default argument state when using the experiment_hparams downstream.

Reply:
Sorry I'm not sure I follow, could you give an example?

Reply:
```
def mutable_default_example(x=[]):
    x.append(1)
    return sum(x)

print(mutable_default_example()) # returns 1
print(mutable_defualt_example()) # returns 2
```